q2_types/feature_data/_format.py

```diff
@@ -100,34 +100,51 @@ class TSVTaxonomyFormat(model.TextFileFormat):
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_tsv_tax_format(self, root, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
+
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for iter, line in file_:
+                iter = iter + 1
 
                 if line == '':
                     # EOF
                     break
-                elif line.lstrip(' ') == '\n':
-                    # Blank line
+                elif line.strip(' ') == '\n':
                     continue
                 elif line.startswith('#'):
                     # Comment line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("Anthony TSVTaxonomy")
                     header = cells
                 else:
                     if len(cells) != len(header):
-                        return False
+                        raise ValidationError("Number of headers are not the "
```

Comment:
:+1: . How about we update this error message to show users what headers we expect to see, and what are the values in the cell we are currently seeing. Also, the error message should show the line number for the line in question.

Reply:
I checked the number of items in the columns against the header values. Should I provide the values for each column to the user? Or just provide the count of items? My last commit checks the count.

Reply:
> show users what headers we expect to see, and what are the values in the cell we are currently seeing

Give the people what they want: show them _the things_, not just the counts.